File: app/src/main/java/com/OverSadBoy/samurairise/view/activity/ItemsAdapterListener.java
package com.OverSadBoy.samurairise.view.activity;

import com.OverSadBoy.samurairise.model.database.Item;

public interface ItemsAdapterListener {
    void onSwitchClick(Item item, boolean status,int position);

    void onItemClick(Item item,int position);

    void onItemLongClick(Item item,int position);
}

File: app/src/main/java/com/OverSadBoy/samurairise/view/activity/ItemsAdapterListener.kt
package com.OverSadBoy.samurairise.view.activity

import androidx.test.runner.AndroidJUnit4
import com.OverSadBoy.samurairise.model.database.Item
import org.junit.runner.RunWith

interface ItemsAdapterListener {
    open fun onSwitchClick(item: Item?, status: Boolean, position: Int)
    open fun onItemClick(item: Item?, position: Int)
    open fun onItemLongClick(item: Item?, position: Int)
}